### PR TITLE
widget fixing

### PR DESCRIPTION
**Pull request by tenisakb**

### CODE REVIEW

File: src/components/shared/currency-widget/currency-widget-slide.tsx [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+const Header = styled.div``;
+
+const Logo = styled.img`
+  width: 64px;
+  height: 64px;
+`;
+
+const HeaderData = styled.div`
+  float: right;
```
Comment (EdmundMai) [inline src/components/shared/currency-widget/currency-widget-slide.tsx]: can we accomplish this with just flexbox?

display: flex;
flex-direction: row-reverse;

File: src/components/shared/currency-widget/currency-widget-slide.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+`;
+
+const Content = styled.div`
+  border-top: 1px solid #e1e5ea;
+  clear: both;
+`;
+
+const ContentRank = styled.div`
+  text-align: center;
+  float: left;
```
Comment (EdmundMai) [inline src/components/shared/currency-widget/currency-widget-slide.tsx]: same question

File: src/components/shared/currency-widget/currency-widget-slide.tsx [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  width: 33%;
+  font-size: 12px;
+  padding: 12px 0;
+  border-right: 1px solid #e1e5ea;
+`;
+
+const ContentVolume = styled.div`
+  text-align: center;
+  float: left;
```
Comment (EdmundMai) [inline src/components/shared/currency-widget/currency-widget-slide.tsx]: same

File: src/components/shared/currency-widget/currency-widget-slide.tsx [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+  width: 33%;
+  font-size: 12px;
+  padding: 12px 0 16px 0;
+  line-height: 1.25em;
+`;
+
+const ContentMarketCap = styled.div`
+  text-align: center;
+  float: left;
```
Comment (EdmundMai) [inline src/components/shared/currency-widget/currency-widget-slide.tsx]: same

File: src/redux/actions/coinmarketcap-action.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+const ACTIONS = Object.freeze({
+  // coinMarketCap download
+  COINMARKETCAP_DOWNLOAD,
+  COINMARKETCAP_SUCCESS,
+  COINMARKETCAP_ERROR,
+
+  // coinMarketCap actions
+  coinMarketCapDownload: () => ({
+    type: ACTIONS.COINMARKETCAP_DOWNLOAD
```
Comment (EdmundMai) [inline src/redux/actions/coinmarketcap-action.tsx]: just `COINMARKETCAP_DOWNLOAD` works too

File: src/redux/actions/coinmarketcap-action.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  }),
+
+  coinMarketCapSuccess: ({ data }) => ({
+    type: ACTIONS.COINMARKETCAP_SUCCESS,
```
Comment (EdmundMai) [inline src/redux/actions/coinmarketcap-action.tsx]: same

File: src/redux/actions/coinmarketcap-action.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    payload: data
+  }),
+
+  coinMarketCapFailure: ({ error }) => ({
+    type: ACTIONS.COINMARKETCAP_ERROR,
```
Comment (EdmundMai) [inline src/redux/actions/coinmarketcap-action.tsx]: same

File: src/config/index.tsx [hunk 8]
```diff
@@ -50,7 +50,9 @@ export const API = Object.freeze({
   V2_STATUS_PATH: ":3000/api/v2/status",
   V1_ACCOUNTS_PATH: ":3000/api/v1/accounts",
   GAS_PRICE: "https://api.blockcypher.com/v1/eth/main",
-  CHUNKS_PER_REQUEST: 10
+  CHUNKS_PER_REQUEST: 10,
+  COINMARKETCAP:
+    "https://widgets.coinmarketcap.com/v2/ticker/3632/?ref=widget&convert=USD"
```
Comment (EdmundMai) [inline src/config/index.tsx]: don't put this under `API`. it is external so make another constant for it. maybe `export const THIRD_PARTY = { COINMARKETCAP: "https://widgets.coinmarketcap.com/v2/ticker/3632/?ref=widget&convert=USD" }`

File: src/redux/reducers/coinmarketcap-reducer.tsx [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+      return { ...state, data: data };
+    case coinMarketCapActions.COINMARKETCAP_ERROR:
+      return { ...state, data: null };
+
+    default:
+      return state;
+  }
+};
+
+export default downloadReducer;
```
Comment (EdmundMai) [inline src/redux/reducers/coinmarketcap-reducer.tsx]: add tests

File: src/redux/actions/coinmarketcap-action.tsx [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    payload: error
+  })
+});
+
+export default ACTIONS;
```
Comment (EdmundMai) [inline src/redux/actions/coinmarketcap-action.tsx]: add tests

File: src/components/shared/currency-widget/currency-widget-slide.tsx (hunk 1 above)
Comment (tenisakb) [reply src/components/shared/currency-widget/currency-widget-slide.tsx]: fixed

File: src/components/shared/currency-widget/currency-widget-slide.tsx (hunk 2 above)
Comment (tenisakb) [reply src/components/shared/currency-widget/currency-widget-slide.tsx]: fixed

File: src/components/shared/currency-widget/currency-widget-slide.tsx (hunk 4 above)
Comment (tenisakb) [reply src/components/shared/currency-widget/currency-widget-slide.tsx]: fixed

File: src/redux/actions/coinmarketcap-action.tsx (hunk 5 above)
Comment (tenisakb) [reply src/redux/actions/coinmarketcap-action.tsx]: fixed

File: src/redux/actions/coinmarketcap-action.tsx (hunk 6 above)
Comment (tenisakb) [reply src/redux/actions/coinmarketcap-action.tsx]: fixed

File: src/redux/actions/coinmarketcap-action.tsx (hunk 7 above)
Comment (tenisakb) [reply src/redux/actions/coinmarketcap-action.tsx]: fixed

File: src/config/index.tsx (hunk 8 above)
Comment (tenisakb) [reply src/config/index.tsx]: fixed

File: src/redux/reducers/coinmarketcap-reducer.tsx (hunk 9 above)
Comment (tenisakb) [reply src/redux/reducers/coinmarketcap-reducer.tsx]: added

File: src/redux/actions/coinmarketcap-action.tsx (hunk 10 above)
Comment (tenisakb) [reply src/redux/actions/coinmarketcap-action.tsx]: added

**Pull request merged**